src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceFunctions.kt

```diff
@@ -20,22 +20,45 @@ data class SentenceChanges(
-  val tussdChanged =
-    nullableDatesDiffer(newSentence.topupSupervisionStartDate, this.topupSupervisionStartDate)
-  val tusedChanged =
-    nullableDatesDiffer(newSentence.topupSupervisionExpiryDate, this.topupSupervisionExpiryDate)
-  val prrdChanged = nullableDatesDiffer(newSentence.postRecallReleaseDate, this.postRecallReleaseDate)
-  val hdcadChanged = this is HdcLicence &&
-    nullableDatesDiffer(
-      newSentence.homeDetentionCurfewActualDate,
-      this.homeDetentionCurfewActualDate,
-    )
+fun SentenceDetail.toSentenceDates() = SentenceDates(
+  conditionalReleaseDate = conditionalReleaseOverrideDate ?: conditionalReleaseDate,
+  actualReleaseDate = confirmedReleaseDate,
+  sentenceStartDate = sentenceStartDate,
+  sentenceEndDate = sentenceExpiryOverrideDate ?: sentenceExpiryDate,
+  licenceExpiryDate = licenceExpiryOverrideDate ?: licenceExpiryDate,
+  topupSupervisionStartDate = topupSupervisionStartDate,
+  topupSupervisionExpiryDate = topupSupervisionExpiryOverrideDate ?: topupSupervisionExpiryDate,
+  postRecallReleaseDate = postRecallReleaseOverrideDate ?: postRecallReleaseDate,
+  homeDetentionCurfewEligibilityDate = homeDetentionCurfewEligibilityDate,
+  homeDetentionCurfewActualDate = homeDetentionCurfewActualDate,
+  homeDetentionCurfewEndDate = homeDetentionCurfewEndDate,
+)
+
+data class SentenceDates(
+  val conditionalReleaseDate: LocalDate? = null,
+  val actualReleaseDate: LocalDate? = null,
+  val sentenceStartDate: LocalDate? = null,
+  val sentenceEndDate: LocalDate? = null,
+  val licenceExpiryDate: LocalDate? = null,
+  val topupSupervisionStartDate: LocalDate? = null,
+  val topupSupervisionExpiryDate: LocalDate? = null,
+  val postRecallReleaseDate: LocalDate? = null,
+  val homeDetentionCurfewActualDate: LocalDate? = null,
+  val homeDetentionCurfewEndDate: LocalDate? = null,
+  val homeDetentionCurfewEligibilityDate: LocalDate? = null,
+)
+
+fun Licence.getSentenceChanges(sentenceDates: SentenceDates, newLsd: LocalDate?): SentenceChanges {
+  val lsdChanged = licenceStartDate.hasChanged(newLsd)
+  val ledChanged = licenceExpiryDate.hasChanged(sentenceDates.licenceExpiryDate)
+  val sedChanged = sentenceEndDate.hasChanged(sentenceDates.sentenceEndDate)
+  val tussdChanged = topupSupervisionStartDate.hasChanged(sentenceDates.topupSupervisionStartDate)
+  val tusedChanged = topupSupervisionExpiryDate.hasChanged(sentenceDates.topupSupervisionExpiryDate)
+  val prrdChanged = postRecallReleaseDate.hasChanged(sentenceDates.postRecallReleaseDate)
+  val hdcadChanged =
+    this is HdcLicence && homeDetentionCurfewActualDate.hasChanged(sentenceDates.homeDetentionCurfewActualDate)
   val hdcEndDateChanged =
-    this is HdcLicence && nullableDatesDiffer(newSentence.homeDetentionCurfewEndDate, this.homeDetentionCurfewEndDate)
+    this is HdcLicence && homeDetentionCurfewEndDate.hasChanged(sentenceDates.homeDetentionCurfewEndDate)
```

Comment:
I know this is existing logic and it's not currently something we need to worry about as in context it's only ever used for pre-release licences, but it may be worth us remembering that this won't be true for HDC Variations, so we can't rely on this function to tell us if the HDC dates have changed in NOMIS for a post-release case. There's no scenario where we need to right now that I can think of, but I thought it worth raising so it wasn't just in my head!